pkgs/by-name/un/unblob/package.nix

```diff
@@ -0,0 +1,103 @@
+{
+  lib,
+  callPackage,
+  python3,
+  fetchFromGitHub,
+  gitUpdater,
+  makeWrapper,
+  e2fsprogs,
+  jefferson,
+  lz4,
+  lziprecover,
+  lzop,
+  p7zip,
+  sasquatch,
+  sasquatch-v4be,
+  simg2img,
+  ubi_reader,
+  unar,
+  zstd,
+}:
+
+let
+  # These dependencies are only added to PATH
+  runtimeDeps = [
+    e2fsprogs
+    jefferson
+    lziprecover
+    lzop
+    p7zip
+    sasquatch
+    sasquatch-v4be
+    ubi_reader
+    simg2img
+    unar
+    zstd
+    lz4
+  ];
+in
+python3.pkgs.buildPythonApplication rec {
+  pname = "unblob";
+  version = "24.9.30";
+  format = "pyproject";
+
+  src = fetchFromGitHub {
+    owner = "onekey-sec";
+    repo = "unblob";
+    rev = version;
+    hash = "sha256-r5UGoCOzYO/v7cAkbhOmeX7caspues/k/gokuzh1wsU=";
+    forceFetchGit = true;
+    fetchLFS = true;
+  };
+
+  strictDeps = true;
+  # jefferson and ubi_reader is installed via pip upstream, but they are not required in the python env
+  dontCheckRuntimeDeps = true; # jefferson and ubi_reader is installed via pip upstream, but they are not required in the python env
+  buildInputs = with python3.pkgs; [ poetry-core ];
+
+  propagatedBuildInputs = with python3.pkgs; [
+    arpy
+    attrs
+    click
+    cryptography
+    dissect-cstruct
+    lark
+    lief.py
+    python3.pkgs.lz4 # shadowed by pkgs.lz4
+    plotext
+    pluggy
+    pyfatfs
+    pyperscan
+    python-magic
+    rarfile
+    rich
+    structlog
+    treelib
+    unblob-native
+  ];
+
+  nativeBuildInputs = [
+    makeWrapper
+  ];
+
+  pythonImportsCheck = [ "unblob" ];
+
+  makeWrapperArgs = [
+    "--prefix PATH : ${lib.makeBinPath runtimeDeps}"
+  ];
+
+  doCheck = false;
+  passthru = {
+    tests.pytest = callPackage ./tests.nix { };
```

Comment:
Inline as using `checkPhase` of the original derivation or inline as putting the test derivation inline in `passthru`?

I liked the approach that I can work on the derivation without the need of running tests.

Reply:
I don't know who wrote this but for pythonPackages we run python tests inside the normal build and only move them out if they cause an infinite recursion because otherwise they are not run when doing python updates to related packages and the failures might not be catched.